uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -498,16 +498,20 @@ define([
     function initRoles () {
       role = 'staff';
 
+      if (vm.isSelfRecord) {
```

Comment:
We should not really change the `role` based on `isSelfRecord`. Because they are mutually exclusive, in future we might have a requirement where the my leave page needs to have different functionality based on the role(staff or manager). So we should keep the role as it is, but add conditions based on `isSelfRecord` to achieve the same.

Reply:
Sure. I am following the similar logic as per Leave Calendar:

```js
bindings: {
  contactId: '<',
  roleOverride: '@?'
}
```
```js
function setUserRole () {
  if (vm.roleOverride) {
    return $q.resolve().then(function () {
      userRole = vm.roleOverride;
    });
  } else {
    return $q.all([
      checkPermissions(sharedSettings.permissions.admin.administer),
      checkPermissions(sharedSettings.permissions.ssp.manage)
    ])
    .then(function (results) {
      userRole = results[0] ? 'admin' : (results[1] ? 'manager' : 'staff');
    });
  }
}
```
And here is the config file for the **my-leave** module:
```js
.state('my-leave.calendar', {
  url: '/calendar',
  template: '<leave-calendar contact-id="myleave.contactId" role-override="staff"></leave-calendar>'
});
```
I actually wanted initially to override the role basing on the `$rootScope.section`.